src/main/java/uk/gov/hmcts/darts/cases/helper/AdvancedSearchRequestHelper.java

```diff
@@ -138,10 +138,18 @@ private List<Predicate> addCourthouseCriteria(GetCasesSearchRequest request, Cri
         List<Predicate> predicateList = new ArrayList<>();
 
         //add courthouse permissions
-        List<Integer> courthouseIdsUserHasAccessTo = authorisationApi.getListOfCourthouseIdsUserHasAccessTo();
+        List<Integer> allCourthouses = authorisationApi.getListOfCourthouseIdsUserHasAccessTo();
+        List<Integer> courthousesIfInterpreter = authorisationApi.getListOfCourthouseIdsUserHasAccessToIfInterpreterUsed();
+        List<Integer> courthousesIgnoringInterpreter = new ArrayList<>(CollectionUtils.disjunction(allCourthouses, courthousesIfInterpreter));
```

Comment:
hi Chris 👋  correct me if I'm wrong but in my particular case I think disjunction will work. 
The reason is that the second set is always a subset of the second. i.e. there will not be a `courthousesIfInterpreter` that doesn't exist in `allCourthouses` (obviously).
So we will have say A that has (1,2,3) and B that has (1,2), with the disjunction being 1, which is the result I'm looking for.

Even if I wanted to change this to your logic, I see two issues:
1)  CollectionUtils.removeAll will throw an exception, because we are working with immutable collections implementations (from Hibernate). This is the reason why I went for `CollectionUtils.disjunction`.
2) your approach won't work in my case, because to make global access work I need to start from the set of ALL possible courthouses user can access. And I can't get that from the security groups. That's why I'm calling `authorisationApi.getListOfCourthouseIdsUserHasAccessTo()`

Happy to discuss this in a call, it's a tricky one